File: pubnub-gson/src/main/java/com/pubnub/api/endpoints/presence/SetState.java
package com.pubnub.api.endpoints.presence;

import com.pubnub.api.endpoints.Endpoint;
import com.pubnub.api.models.consumer.presence.PNSetStateResult;

public interface SetState extends Endpoint<PNSetStateResult> {
    SetState channels(java.util.List<String> channels);

    SetState channelGroups(java.util.List<String> channelGroups);

    SetState state(Object state);

    SetState uuid(String uuid);

    SetState withHeartbeat(boolean withHeartbeat);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/kmp/endpoints/presence/SetState.kt
package com.pubnub.kmp.endpoints.presence

import com.pubnub.kmp.PNFuture
import com.pubnub.api.models.consumer.presence.PNSetStateResult

/**
 * @see [PubNub.setPresenceState]
 */
interface SetState : PNFuture<PNSetStateResult> {
}